GaeManda/Projects/Presentation/GMDProfile/Implementations/DogProfileEdit/DogProfileEditViewController.swift

```diff
@@ -0,0 +1,388 @@
+			UIImage(systemName: "person.crop.circle")
+		]
+		
+		Observable<[UIImage?]>.of(images)
+			.bind(to: dogProfileDashBoard.rx.items(
+				cellIdentifier: DogProfileDashBoardCell.identifier,
+				cellType: DogProfileDashBoardCell.self
+			)) { (index, item, cell) in
+				cell.configuration(image: item)
+				if index == 1 {
+					cell.isEdited = true
+				}
+			}
+			.disposed(by: disposeBag)
+		
+		scrollViewTextFieldBind()
+		scrollViewButtonBind()
+	}
+	
+	/// ScrollView TextField Bind
+	func scrollViewTextFieldBind() {
+		scrollView.nickNameTextField.textField.rx.text
+			.orEmpty
+			.withUnretained(self)
+			.bind { owner, text in
+				owner.setTextCountLabel(text)
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.calenderTextField.textField.rx.controlEvent(.editingDidBegin)
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.scrollView.calenderTextField.textField.endEditing(true)
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.weightTextField.textField.rx.controlEvent(.editingChanged)
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.addSuffixForWeightTextField()
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.weightTextField.textField.rx.cursorChanged
+			.withUnretained(self)
+			.bind { owner, range in
+				owner.setUneditableSuffix(range)
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.characterTextView.textView.rx.text
+			.orEmpty
+			.map { $0.count }
+			.withUnretained(self)
+			.bind { owner, count in
+				owner.setTextCountLabel(count)
+			}
+			.disposed(by: disposeBag)
+	}
+	
+	/// ScrollView Button Action Bind
+	func scrollViewButtonBind() {
+		scrollView.calenderButton.rx.tap
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.calenderButtonDidTap()
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.maleButton.rx.tap
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.maleButtonDidTap()
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.femaleButton.rx.tap
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.femaleButtonDidTap()
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.didNeuterButton.rx.tap
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.didNeuterButtonDidTap()
+			}
+			.disposed(by: disposeBag)
+		
+		scrollView.didNotNeuterButton.rx.tap
+			.withUnretained(self)
+			.bind { owner, _ in
+				owner.didNotNeuterButtonDidTap()
+			}
+			.disposed(by: disposeBag)
+	}
+}
+
+// MARK: Action
+private extension DogProfileEditViewController {
+	@objc func backbuttonDidTap() {
+		listener?.backbuttonDidTap()
+	}
+	
+	func setTextCountLabel(_ text: String) {
+		var newText = text
+		let textCount = scrollView.maximumTextFieldCount
+		
+		if text.count >= textCount {
+			let index = text.index(text.startIndex, offsetBy: textCount)
+			newText = String(text[..<index])
+			scrollView.nickNameTextField.textField.text = newText
+		}
+		scrollView.maximumTextCountLabel.text = "\(newText.count)/\(textCount)"
```

Comment:
레이블의 값을 직접 업데이트하는 코드와 rx로 업데이트하는 코드가 동시에 있는데!
레이블의 값을 업데이트를 추적할 때 복잡해져서 권장하지 않음!! rx스럽게 바꿔보는 것 어때!! 대충 아래처럼 ㅇㅅㅇ 대충짜봄ㅋㅋ

```suggestion

nickNameTextField.textField.rx.text
  .orEmpty
  .map { text -> String in
    if text.count >= maximumTextCount {
      let index = text.index(text.startIndex, offsetBy: maximumTextCount)
      return String(text[..<index])
    }
    return text
  }
  .bind(to: nickNameTextField.textField.rx.text)
  .disposed(by: disposeBag)
        
  nickNameTextField.textField.rx.text
    .orEmpty
    .map { "\($0.count)/\(maximumTextCount)" }
    .bind(to: maximumTextCountLabel.rx.text)
    .disposed(by: disposeBag)
```